pkg/sql/distsqlrun/processors.go

```diff
@@ -500,7 +500,7 @@ func (a AggregatorSpec_Aggregation) Equals(b AggregatorSpec_Aggregation) bool {
 			return false
 		}
 	} else {
-		if a.FilterColIdx == nil || *a.FilterColIdx != *b.FilterColIdx {
+		if b.FilterColIdx == nil || *a.FilterColIdx != *b.FilterColIdx {
```

Comment:
This was a neat little bug that surfaced.

Reply:
Consider putting this and any associated test in a separate commit (would be easier to look at if someone looks at code history)